### PR TITLE
fix(permission): optimize permission calculation logic in dv

### PR DESCRIPTION
**Pull request by ybzky**
close #xxx

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/sheets-data-validation/src/views/list/index.tsx [hunk 1]
```diff
@@ -95,22 +94,29 @@ function DataValidationListWithWorkbook(props: { workbook: Workbook }) {
     const getDvRulesByPermissionCorrect = (rules: ISheetDataValidationRule[]) => {
```
Comment (Gggpound) [inline packages/sheets-data-validation/src/views/list/index.tsx:94]: 这个函数在react组件每次渲染的时候 都要跑一次？不需要memo么？

Comment (ybzky) [reply packages/sheets-data-validation/src/views/list/index.tsx:94]: 不需要 看了这个组件只会rule变化的时候渲染 rule变化本身就要重新算的

File: packages/sheets-data-validation/src/views/list/index.tsx [hunk 2]
```diff
@@ -95,22 +94,29 @@ function DataValidationListWithWorkbook(props: { workbook: Workbook }) {
...
         const workbook = univerInstanceService.getCurrentUnitForType<Workbook>(UniverInstanceType.UNIVER_SHEET)!;
         const worksheet = workbook.getActiveSheet();
+        const unitId = workbook.getUnitId();
```
Comment (wzhudev) [inline packages/sheets-data-validation/src/views/list/index.tsx:97]: 这里否是可以抽一个 hooks 出来？

File: packages/sheets-data-validation/src/views/list/index.tsx [hunk 3]
```diff
@@ -95,22 +94,29 @@ function DataValidationListWithWorkbook(props: { workbook: Workbook }) {
...
-                for (let row = startRow; row <= endRow; row++) {
-                    for (let col = startColumn; col <= endColumn; col++) {
-                        const permission = (worksheet?.getCell(row, col) as (ICellDataForSheetInterceptor & { selectionProtection: ICellPermission[] }))?.selectionProtection?.[0];
-                        if (permission?.[UnitAction.Edit] === false || permission?.[UnitAction.View] === false) {
-                            return true;
-                        }
-                    }
-                }
-                return false;
+            const lapPermission = permissions.find((permission) => {
```
Comment (wzhudev) [inline packages/sheets-data-validation/src/views/list/index.tsx:110]: overlap

File: packages/sheets-data-validation/src/views/list/index.tsx [hunk 4]
```diff
@@ -95,22 +94,29 @@ function DataValidationListWithWorkbook(props: { workbook: Workbook }) {
...
+                return ranges?.some((range) => {
```
Comment (wzhudev) [inline packages/sheets-data-validation/src/views/list/index.tsx:111]: 这个循环方法也可以抽一个通用函数

**Review by wzhudev — APPROVED** [review]
LGTM

**Pull request merged**